packages/sit-onyx/src/utils/playwright.tsx

```diff
@@ -0,0 +1,177 @@
+import type { JSX } from "vue/jsx-runtime";
+import { expect, test } from "../playwright-axe";
+
+type Permutation<T extends Record<string, readonly string[]>> = {
+  [key in keyof T]: T[key][number];
+};
+
+/**
+ * recursive generate permutations function
+ */
+const _generatePermutations = <
+  T extends Readonly<Record<string, readonly string[]>>,
+  K extends keyof T,
+  R extends Permutation<T>,
+>(
+  obj: T,
+  keys: K[],
+  current: Partial<R> = {},
+  result: R[] = [],
+): R[] => {
+  if (keys.length === 0) {
+    result.push({ ...current } as R);
+  } else {
+    const [key, ...restKeys] = keys;
+    const values = obj[key];
+    for (const value of values) {
+      current[key] = value as R[K];
+      _generatePermutations(obj, restKeys, current, result);
+    }
+  }
+  return result;
+};
+
+/**
+ * Creates an array of all possible permutations of the values of an object
+ */
+const generatePermutations = <T extends Record<string, readonly string[]>>(obj: T) => {
+  const keys = Object.keys(obj) as (keyof T)[];
+  return _generatePermutations(obj, keys);
+};
+
+/**
+ * Builds a grid-template-area value for the screenshot matrix test.
+ *
+ * @example
+ * ```
+ * ". rowHeadline0 rowHeadline1 rowHeadline2"
+ * "colHeadline0 case0 case1 case2"
+ * "colHeadline1 case3 case4 case5"
+ * "colHeadline2 case6 case7 case8"
+ * "colHeadline3 case9 case10 case11"
+ * ```
+ */
+const buildGridTemplate = (rows: number, columns: number) =>
+  Array.from({ length: rows + 1 })
+    .map((_, ri) =>
+      Array.from({ length: columns + 1 })
+        .map((_, ci) => {
```

Comment:
Would prefer to call the variables "row" and "column" so its easier understandable what they are